docs/development.md

```diff
@@ -1,22 +1,44 @@
 # Development
+- [Development](#development)
+  - [Required Versions:](#required-versions)
+  - [Mac Prerequisites](#mac-prerequisites)
+  - [Linux Prerequisites](#linux-prerequisites)
+  - [Local Development](#local-development)
 
-- Install cmake or (optionally, required tools for GPUs)
-- run `go generate ./...`
-- run `go build .`
-
-Install required tools:
+## Required Versions:
 
 - cmake version 3.24 or higher
 - go version 1.20 or higher
 - gcc version 11.4.0 or higher
 
+
+## Mac Prerequisites
+
 ```bash
 brew install go cmake gcc
 ```
 
+## Linux Prerequisites
+
+- Install cmake and gcc
+  - run `sudo apt-get install cmake gcc`
```

Comment:
apt is not used by all Linux distributions

maybe: either add alternatives, remove specific install commands, or specify version?